spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/partition/DeployerPartitionHandler.java

```diff
@@ -329,11 +349,22 @@ private void launchWorker(StepExecution workerStepExecution) {
 
 		AppDefinition definition = new AppDefinition(resolveApplicationName(),
 				environmentVariables);
-
+		TaskExecution taskExecution = this.taskRepository.createTaskExecution();
 		AppDeploymentRequest request = new AppDeploymentRequest(definition, this.resource,
-				this.deploymentProperties, arguments);
+				this.deploymentProperties, commandLineArgsPrep(taskExecution, arguments));
+
+		String externalExecutionId = this.taskLauncher.launch(request);
+		this.taskRepository.updateExternalExecutionId(taskExecution.getExecutionId(),
+				externalExecutionId);
+	}
 
-		this.taskLauncher.launch(request);
+	private List<String> commandLineArgsPrep(TaskExecution taskExecution,
```

Comment:
We let the user choose if the args we provide are passed via command line args or env vars.  This property should be treated the same.  See lines 313 and 334.

Reply:
Done